app/services/atmosphere/billing_service.rb

```diff
@@ -20,45 +20,65 @@ def self.bill_all_appliances
     end
 
     # This method effects billing for a single appliance.
-    # message is written to billing logs
-    # apply_prepayment determines whether the appliance should be prepaid (false when shutting down appliance)
+    # It works by iterating over this appliance's deployments and billing each one separately
+    # A message is written to billing logs.
+    # apply_prepayment determines whether the deployment should be prepaid
+    # (false when shutting down appliance or deployment)
     def self.bill_appliance(appl, billing_time, message, apply_prepayment = true)
-      if appl.prepaid_until.blank?
-        # This should not happen - it means that we do not know when this appliance will run out of funds. Therefore we cannot meaningfully bill it again and must return an error.
-        Rails.logger.error("Unable to determine current payment validity date for appliance #{appl.id}. Skipping.")
-        BillingLog.new(timestamp: Time.now.utc, user: appl.appliance_set.user, appliance: appl.id.to_s+'---'+(appl.name.blank? ? 'unnamed_appliance' : appl.name), fund: appl.fund.name, message: "Unable to determine current payment validity date for appliance.", actor: "bill_appliance", amount_billed: 0).save
-        appl.billing_state = "error"
-        appl.save
-        raise Atmosphere::BillingException.new(message: "Unable to determine current payment validity date for appliance #{appl.id}")
+      appl.deployments.each do |dep|
+        bill_deployment(dep, billing_time, message, apply_prepayment)
+      end
+    end
+
+
+    def self.bill_deployment(dep, billing_time, message, apply_prepayment = true)
+      appl = dep.appliance # Shorthand
+      if dep.prepaid_until.blank?
```

Comment:
It is not for now, but future PR: this condition smell to me like model validation. Maybe it can be moved there? I create dedicated issue to track this issue: #106